py_ecc/fields/field_elements.py

```diff
@@ -0,0 +1,377 @@
+        elif isinstance(other, FQP):
+            return self * other.inv()
+        else:
+            raise TypeError(
+                "Expected an int or FQ object or FQP object, but got object of type {}"
+                .format(type(other))
+            )
+
+    def __truediv__(self, other: Union[int, "FQ", "FQP"]) -> "FQP":
+        return self.__div__(other)
+
+    def __pow__(self, other: int) -> "FQP":
+        if other == 0:
+            return type(self)([1] + [0] * (self.degree - 1))
+        elif other == 1:
+            return type(self)(self.coeffs)
+        elif other % 2 == 0:
+            return (self * self) ** (other // 2)
+        else:
+            return ((self * self) ** int(other // 2)) * self
+
+    # Extended euclidean algorithm used to find the modular inverse
+    def inv(self) -> "FQP":
+        lm, hm = (
+            [1] + [0] * self.degree,
+            [0] * (self.degree + 1),
+        )
+        low, high = (
+            # Ignore mypy yelling about the inner types for  the tuples being incompatible
+            cast(List[IntOrFQ], list(self.coeffs + (0,))),  # type: ignore
+            cast(List[IntOrFQ], list(self.modulus_coeffs + (1,))),  # type: ignore
+        )
+        while deg(low):
+            r = cast(List[IntOrFQ], list(poly_rounded_div(high, low)))
+            r += [0] * (self.degree + 1 - len(r))
+            nm = [x for x in hm]
+            new = [x for x in high]
+
+            if len(lm) != self.degree + 1:
+                raise Exception("Length of lm is not {}".format(self.degree + 1))
+            elif len(hm) != self.degree + 1:
+                raise Exception("Length of hm is not {}".format(self.degree + 1))
+            elif len(nm) != self.degree + 1:
+                raise Exception("Length of nm is not {}".format(self.degree + 1))
+            elif len(low) != self.degree + 1:
+                raise Exception("Length of low is not {}".format(self.degree + 1))
+            elif len(high) != self.degree + 1:
+                raise Exception("Length of high is not {}".format(self.degree + 1))
+            elif len(new) != self.degree + 1:
+                raise Exception("Length of new is not {}".format(self.degree + 1))
+
+            for i in range(self.degree + 1):
+                for j in range(self.degree + 1 - i):
+                    nm[i + j] -= lm[i] * int(r[j])
+                    new[i + j] -= low[i] * int(r[j])
+            lm, low, hm, high = nm, new, lm, low
+        return type(self)(lm[:self.degree]) / low[0]
+
+    def __repr__(self) -> str:
+        return repr(self.coeffs)
+
+    def __eq__(self, other: "FQP") -> bool:     # type: ignore # https://github.com/python/mypy/issues/2783 # noqa: E501
+        if not isinstance(other, type(self)):
+            raise TypeError(
+                "Expected an FQP object, but got object of type {}"
+                .format(type(other))
+            )
+
+        for c1, c2 in zip(self.coeffs, other.coeffs):
```

Comment:
I *think* that since we have the `isinstance` check up top we're safe here, but worth pointing out that `zip` discards extra values if one of the values is longer so unless we know that these are the same length there's an error edge cases here.

Reply:
@pipermerriam https://github.com/ethereum/py_ecc/issues/45 should take care of this. (Probably a separate PR for this.)